### PR TITLE
Feature/nt 0024

### PR DESCRIPTION
**Pull request by damian0o**
Work in progress. Just for peers notification.

### CODE REVIEW

**Comment by damian0o** [comment]
> Work in progress. Just for peers notification.

Ready for review

File: client/src/containers/SignUp/SignUpForm/SignUpForm.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        />
+        <Field
+          name="passwordRepeat"
+          type="password"
+          error={!!errors.passwordRepeat}
+          label="Repeat Password*"
+          outllined
+          component={TextField}
+        />
+        <FormErrors errors={errors} />
```
Comment (kamilmoskal) [inline client/src/containers/SignUp/SignUpForm/SignUpForm.tsx]: This component should be displayed conditionally becouse if there are no errors, it showing white space with margin i guess

File: client/src/store/actions/authActions.tsx [hunk 2]
```diff
@@ -10,6 +10,14 @@ export interface LoginData {
   remember: boolean;
 }
 
+export interface RegisterData {
```
Comment (kamilmoskal) [inline client/src/store/actions/authActions.tsx]: We have this in authTypes.tsx too, so we should remove this and import one from authTypes, same for LoginData

File: client/src/store/actions/authActions.tsx [hunk 3]
```diff
@@ -31,3 +39,28 @@ export const loginUser = (
...
+    })
+    .catch(err => {
+      dispatch({
+        type: TYPES.REGISTER_ERROR,
+        errors: err.response.data.errors
+      });
+    });
+};
+
+export const registerStartLoading = () => ({ type: TYPES.REGISTER_LOADING });
```
Comment (kamilmoskal) [inline client/src/store/actions/authActions.tsx]: this line is unnecessary, same for authStartLoading, becouse we dispatching type directly inside login/register function

File: client/src/store/interfaces/authTypes.tsx [hunk 4]
```diff
@@ -23,4 +31,23 @@ export interface LoginErrorAction {
...
+}
+export interface RegisterErrorAction {
+  type: "REGISTER_ERROR";
+  errors: [{ msg: string }];
+}
+export interface RegisterLoadingAction {
+  type: "REGISTER_LOADING";
+}
+
+export type Action =
```
Comment (kamilmoskal) [inline client/src/store/interfaces/authTypes.tsx]: Maybe better if we divide this on LoginActions / RegisterActions

File: client/src/containers/LandingPage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const LandingPage: React.FC = () => {
+    return (
+        <div>NaTrepa Portal Landing Page</div>
```
Comment (mras2303) [inline client/src/containers/LandingPage/index.tsx]: NaTrepa?

File: client/src/containers/SignUp/SignUpForm/SignUpForm.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    onSubmit={values => {
+      onSubmit(values);
+    }}
+    validationSchema={SignupSchema}
+  >
+    {({ errors }) => (
+      <StyledFormikForm>
+        <Field
+          name="firstName"
+          error={!!errors.firstName}
```
Comment (mras2303) [inline client/src/containers/SignUp/SignUpForm/SignUpForm.tsx]: Is that correct? !!value just return true or false

Comment (kamilmoskal) [reply client/src/containers/SignUp/SignUpForm/SignUpForm.tsx]: Yes, the Field component expects boolean value from error prop to highlight 'input' on red

File: client/src/containers/LandingPage/index.tsx [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const LandingPage: React.FC = () => {
+    return (
+        <div>NaTripa Portal Landing Page</div>
```
Comment (mras2303) [inline client/src/containers/LandingPage/index.tsx]: Please make any link to login/register form. I don't want to type it in URL ;)

File: client/src/containers/SignUp/SignUpForm/SignUpForm.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+          </p>
+        ))}
+      </TextHelper>
+    );
+  } else {
+    return (<></>)
+  }
+};
+
+interface Values {
```
Comment (mras2303) [inline client/src/containers/SignUp/SignUpForm/SignUpForm.tsx]: It looks the same like in file `SignUp.tsx `

Can we export it to external file and import in both files?

File: client/src/store/actions/authActions.tsx [hunk 9]
```diff
@@ -30,4 +24,25 @@ export const loginUser = (
...
-export const authStartLoading = () => ({ type: TYPES.LOGIN_LOADING });
+export const registerUser = (
+  registerData: RegisterData
+): ThunkAction<void, {}, {}, RegisterAction> => async dispatch => {
+  dispatch({
+    type: TYPES.REGISTER_LOADING
+  });
+
+  axios
+    .post("/api/users/register", registerData)
```
Comment (mras2303) [inline client/src/store/actions/authActions.tsx]: FUTURE

We need to talk about axios.base method

File: client/src/store/actions/authActions.tsx [hunk 10]
```diff
@@ -30,4 +24,25 @@ export const loginUser = (
...
+    .then(resp => {
```
Comment (mras2303) [inline client/src/store/actions/authActions.tsx]: We can destructure this value

`.then({ data } => {`

File: client/src/containers/SignUp/SignUpForm/SignUpForm.tsx (hunk 8 above)
Comment (damian0o) [reply client/src/containers/SignUp/SignUpForm/SignUpForm.tsx]: Done

File: client/src/store/actions/authActions.tsx (hunk 10 above)
Comment (damian0o) [reply client/src/store/actions/authActions.tsx]: Done

File: client/src/containers/LandingPage/index.tsx (hunk 5 above)
Comment (damian0o) [reply client/src/containers/LandingPage/index.tsx]: Fixed

File: client/src/store/interfaces/authTypes.tsx (hunk 4 above)
Comment (damian0o) [reply client/src/store/interfaces/authTypes.tsx]: Done

File: client/src/store/actions/authActions.tsx (hunk 3 above)
Comment (damian0o) [reply client/src/store/actions/authActions.tsx]: Removed

File: client/src/store/actions/authActions.tsx (hunk 2 above)
Comment (damian0o) [reply client/src/store/actions/authActions.tsx]: Removed

File: client/src/containers/SignUp/SignUpForm/SignUpForm.tsx (hunk 1 above)
Comment (damian0o) [reply client/src/containers/SignUp/SignUpForm/SignUpForm.tsx]: Done

**Comment by kamilmoskal** [comment]
This is not task for navigation and i think we shouldn't overextend next features inside this single one. Navigation should be in diffrent component and imported in diffrent place, not inside LandingPage. Landing Page should display main content but we don't have any mockup at this moment.

**Pull request merged**